pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -421,6 +430,26 @@ pub mod pallet {
 				Err(Error::<T>::NotFound.into())
 			}
 		}
+
+		#[cfg(any(feature = "try-runtime", test))]
+		pub fn do_try_state() -> Result<(), &'static str> {
+			ConnectedDids::<T>::iter().try_for_each(|(account, record)| -> Result<(), &'static str> {
+				ensure!(
+					ConnectedAccounts::<T>::contains_key(record.did, account),
+					"Unknown account"
+				);
+				Ok(())
+			})?;
+
+			ConnectedAccounts::<T>::iter().try_for_each(
+				|(did_identifier, linked_account_id, _)| -> Result<(), &'static str> {
+					let connected_did = ConnectedDids::<T>::get(linked_account_id);
+					ensure!(connected_did.is_some(), "Unknown did");
```

Comment:
This can still be simplified by checking against `Some(...)`.

Reply:
Are you sure? I want to check a property of `connected_did`. 

Doing something like: 
```
connected_did == Some(ConnectionRecord {
					did: did_identifier,
					deposit: deposit
				})
```

looks to me over-engineered.

Reply:
I also do not know the deposit – I needed to copy it from the `connected_did` variable -> I have to unwrap it...

Reply:
I see, then probably calling `.expect(...)` on ConnectedDids::<T>::get()` and then matching the DID is a better option, in my opinion. All cases panic anyway.